### PR TITLE
Add short-term roadmap

### PR DESCRIPTION
**Pull request by eatonphil**
The README does mention the project board but a short-term roadmap may get a little quicker to the next few major points.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,12 @@ Read more about the [history](./docs/HISTORY.md) of TigerBeetle, the problem of
 
 ## TigerBeetle (under active development)
 
-TigerBeetle is not yet production-ready. The production version of **TigerBeetle is now under active development**. Our [DESIGN doc](docs/DESIGN.md) provides an overview of TigerBeetle's data structures and our [project board](https://github.com/tigerbeetledb/tigerbeetle/projects?type=classic) provides a glimpse of where we want to go.
+TigerBeetle is not yet production-ready. The production version of
+**TigerBeetle is now under active development**. Our [DESIGN
+doc](docs/DESIGN.md) provides an overview of TigerBeetle's data
+structures.
+
+Check out short-term Roadmap below for where we're heading!
```
Comment (sentientwaffle) [inline README.md]:
```suggestion
Check out our short-term Roadmap below for where we're heading!
```

And maybe "Short-term Roadmap" could be a link.

File: README.md [hunk 2]
```diff
@@ -233,6 +238,17 @@ You may be interested in:
...
+## Short-Term Roadmap
+
+Done:
+
+1. Harden the consensus layer
+
+In progress:
+
+2. Harden the storage layer (until this happens you'll occasionally see panics in the LSM code)
+3. Support querying an account for its transfers (until this happens you can only query account balances and list all transfers)
```
Comment (sentientwaffle) [inline README.md]: We don't support listing all transfers. You can query accounts or transfers by id ("point queries").

Also, maybe reword "querying an account for its transfers" to "querying for transfers by their debit/credit account". "account for its transfers" sounds kind we are querying the account, and extracting its transfers.

File: README.md (hunk 1 above)
Comment (eatonphil) [reply README.md]: Done and done

File: README.md (hunk 2 above)
Comment (eatonphil) [reply README.md]: Whoops! Thanks

File: README.md [hunk 3]
```diff
@@ -233,6 +238,22 @@ You may be interested in:
...
+
+Done:
+
+1. Consensus layer readiness
+
+In progress:
+
+2. Storage layer readiness
+  * Until this happens you'll occasionally see panics in the LSM code.
+  * See [#189](https://github.com/tigerbeetledb/tigerbeetle/issues/189) for details
```
Comment (jorangreef) [inline README.md]:
```suggestion
  * See [#189](https://github.com/tigerbeetledb/tigerbeetle/issues/189) for details.
```

To use full stops consistently for all bullets.

File: README.md [hunk 4]
```diff
@@ -233,6 +238,22 @@ You may be interested in:
...
+## Short-Term Roadmap
+
+Done:
+
+1. Consensus layer readiness
+
+In progress:
+
+2. Storage layer readiness
+  * Until this happens you'll occasionally see panics in the LSM code.
```
Comment (jorangreef) [inline README.md]: Would be great just to explain why the panics happen. i.e. Assertion failures as we tune overtight or incorrect assertions, or as assertions detect bugs and shut TigerBeetle down safely.

We don't want people to panic when they see that TigerBeetle panics. Rather, to explain this also to build trust.

Comment (jorangreef) [reply README.md]: Over time, assertion failures will become more and more rare, as we increase assurance through testing.

File: README.md [hunk 5]
```diff
@@ -233,6 +238,22 @@ You may be interested in:
 * [demos/io_uring](./demos/io_uring), how ring buffers can eliminate kernel syscalls, reduce server hardware requirements by a factor of two, and change the way we think about event loops.
 * [demos/hash_table](./demos/hash_table), how linear probing compares with cuckoo probing, and what we look for in a hash table that needs to scale to millions (and billions) of account transfers.
 
+## Short-Term Roadmap
+
+Done:
+
+1. Consensus layer readiness
```
Comment (jorangreef) [inline README.md]:
```suggestion
1. Consensus protocol readiness.
```

File: README.md [hunk 6]
```diff
@@ -233,6 +238,22 @@ You may be interested in:
...
+
+In progress:
+
+2. Storage layer readiness
```
Comment (jorangreef) [inline README.md]:
```suggestion
2. Storage engine readiness.
```

**Review by jorangreef — DISMISSED** [review]
LATM with these changes.

File: README.md (hunk 4 above)
Comment (eatonphil) [reply README.md]: Good call, added.

File: README.md [hunk 7]
```diff
@@ -233,6 +238,24 @@ You may be interested in:
...
+In progress:
+
+2. Storage engine readiness.
+  * Until this happens you'll occasionally see assertion failures in
+    LSM code as we tune overtight or incorrect assertions, or as
+    assertions detect bugs and shut TigerBeetle down safely.
+  * See [#189](https://github.com/tigerbeetledb/tigerbeetle/issues/189) for details.
+3. Recovery readiness.
+  * See [#212](https://github.com/tigerbeetledb/tigerbeetle/issues/212) for details.
+3. Support querying all transfers associated with an account id.
```
Comment (sentientwaffle) [inline README.md]:
```suggestion
4. Support querying all transfers associated with an account id.
```

File: README.md [hunk 8]
```diff
@@ -233,6 +238,24 @@ You may be interested in:
...
+## Short-Term Roadmap
+
+Done:
+
+1. Consensus protocol readiness.
+
+In progress:
+
+2. Storage engine readiness.
+  * Until this happens you'll occasionally see assertion failures in
```
Comment (sentientwaffle) [inline README.md:250]:
```suggestion
  * Until this is implemented you'll occasionally see assertion failures in
```

File: README.md [hunk 9]
```diff
@@ -233,6 +238,24 @@ You may be interested in:
...
+    LSM code as we tune overtight or incorrect assertions, or as
+    assertions detect bugs and shut TigerBeetle down safely.
+  * See [#189](https://github.com/tigerbeetledb/tigerbeetle/issues/189) for details.
+3. Recovery readiness.
+  * See [#212](https://github.com/tigerbeetledb/tigerbeetle/issues/212) for details.
+3. Support querying all transfers associated with an account id.
+  * Until this happens you can only query individual accounts and transfers by id.
```
Comment (sentientwaffle) [inline README.md]:
```suggestion
  * Until this is implemented you can only query individual accounts and transfers by id.
```

**Review by jorangreef — APPROVED** [review]
This is great. Nice work @eatonphil !
